lib/iris/tests/integration/test_FieldsFileVariant.py

```diff
@@ -348,6 +348,88 @@ def test_save_packed_as_unpacked(self):
             test_data_new = ffv.fields[0].get_data()
             self.assertArrayAllClose(test_data_old, test_data_new)
 
+    def test_save_packed_unchanged(self):
+        # Check that we can successfully save a packed datafile as packed.
+        src_path = tests.get_data_path(('FF', 'n48_multi_field'))
+        with self.temp_filename() as temp_path:
+            # Make a copy and open for UPDATE (maybe not strictly necessary?)
+            shutil.copyfile(src_path, temp_path)
+            ffv = FieldsFileVariant(temp_path, FieldsFileVariant.UPDATE_MODE)
+            self.assertEqual(ffv.fields[0].lbpack, 1)
+            old_size = ffv.fields[0].lbnrec
+            test_data_old = ffv.fields[0].get_data()
+            ffv.close()
+
+            ffv = FieldsFileVariant(temp_path)
+            self.assertEqual(ffv.fields[0].lbpack, 1)
+            new_size = ffv.fields[0].lbnrec
+            test_data_new = ffv.fields[0].get_data()
+            msg = 'unpacked LBNREC({}) is != packed({})'
+            self.assertEqual(new_size, old_size,
+                             msg=msg.format(new_size, old_size))
+
+            test_data_new = ffv.fields[0].get_data()
+            self.assertArrayAllClose(test_data_old, test_data_new)
+
+    def test_save_packed_mixed(self):
```

Comment:
It would be lovely to start building some unit tests for this, but I accept that is a challenge without factoring out the data writing component (which I suspect you will want to do when implementing packing in um.py).